modulemd/tests/test-modulemd-packager-v3.c

```diff
@@ -402,192 +387,44 @@ packager_test_map_to_stream_v2_autoname (void)
-  g_assert_cmpstr ("Description",
-                   ==,
-                   modulemd_module_stream_v3_get_description (v3_stream, "C"));
-  g_assert_cmpstr (
-    "f33", ==, modulemd_module_stream_v3_get_platform (v3_stream));
-
-  /* a module license is not required for StreamV3 as it is for StreamV2, so
-   * the license list should be an empty
-   */
-  list = modulemd_module_stream_v3_get_module_licenses_as_strv (v3_stream);
-  g_assert_nonnull (list);
-  g_assert_null (list[0]);
-  g_clear_pointer (&list, g_strfreev);
-
-  /* PackagerV3 to Index conversion should automatically generate a module and
-   * stream name so it can be added to an index
-   */
-  index = modulemd_packager_v3_to_stream_v3_ext (packager, &error);
-  g_assert_no_error (error);
-  g_assert_nonnull (index);
-  g_assert_true (MODULEMD_IS_MODULE_INDEX (index));
-
-  /* however, the automatic module/stream names should not appear in an index dump */
-  yaml_str = modulemd_module_index_dump_to_string (index, &error);
-  g_assert_no_error (error);
-  g_assert_nonnull (yaml_str);
-  g_assert_cmpstr (yaml_str,
-                   ==,
-                   "---\n"
-                   "document: modulemd-stream\n"
-                   "version: 3\n"
-                   "data:\n"
-                   "  context: ctx42\n"
-                   "  summary: Summary\n"
-                   "  description: >-\n"
-                   "    Description\n"
-                   "  dependencies:\n"
-                   "    platform: f33\n"
-                   "...\n");
-
-  g_clear_pointer (&yaml_str, g_free);
-  g_clear_object (&v3_stream);
-  g_clear_object (&packager);
-  g_clear_object (&index);
-
-  /* restore default mdversion to avoid unexpected results from other tests */
-  modulemd_set_default_stream_mdversion (default_mdv);
-}
 
 static void
 packager_test_read_to_index (void)
 {
   gboolean ret;
+  guint i;
   g_autoptr (ModulemdModuleIndex) index = NULL;
+  g_autoptr (ModulemdSubdocumentInfo) subdoc = NULL;
   g_autoptr (GError) error = NULL;
   g_autoptr (GPtrArray) failures = NULL;
   g_autofree gchar *yaml_path = NULL;
   g_autofree gchar *yaml_str = NULL;
   g_autofree gchar *expected_path = NULL;
   g_autofree gchar *expected_str = NULL;
-  ModulemdModuleStreamVersionEnum default_mdv;
-
-  /* get and save current default stream mdversion */
-  default_mdv = modulemd_get_default_stream_mdversion ();
 
   /* create a stream v2 index */
-  modulemd_set_default_stream_mdversion (MD_MODULESTREAM_VERSION_TWO);
   index = modulemd_module_index_new ();
 
   /* The modulemd-packager v3 definition */
   yaml_path = g_strdup_printf ("%s/yaml_specs/modulemd_packager_v3.yaml",
                                g_getenv ("MESON_SOURCE_ROOT"));
   ret = modulemd_module_index_update_from_file (
     index, yaml_path, TRUE, &failures, &error);
-  g_assert_true (ret);
   g_assert_no_error (error);
-  g_assert_cmpint (failures->len, ==, 0);
```

Comment:
This assertion should be kept, just done after dumping any failures and asserting `ret` is true.